### PR TITLE
[#159022590] Add asset spec groups

### PR DESCRIPTION
**Pull request by Arthur236**
## What does this PR do?

Enables admin users to add asset specification groups

### Acceptance Criteria

```gherkin
Scenario: adding asset specs
Given an authenticated admin
When I add asset specs
Then I should see the asset specs group on the list of available asset specs
```

## How should this be manually tested?

* Log in
* Click on the `Asset Specs` link on the Dashboard
* Click on the `Add Asset Spec Group` button on the Asset Specs page
* Fill in the necessary details and click on `Save`

## What are the relevant pivotal tracker stories?

[159022590](https://www.pivotaltracker.com/story/show/159022590)

## Any background context you want to add?

N/A

## Important notes

N/A

## Packages installed

N/A

## Deployment note

N/A

## Related PRs branch | PR (branch_name) | (pr_link)

N/A

## Todos
- [x] Raise PR
- [ ] Test
- [ ] Documentation

## Screenshots

screen shot 2018-10-17 at 16 45 29

screen shot 2018-10-17 at 16 45 58

**New Design**
screen shot 2018-10-18 at 15 33 22

### CODE REVIEW

File: src/__test__/_components/AddAssetSpecContainer.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import expect from 'expect';
+import AddAssetContainer from '../../_components/AssetSpecs/AddAssetSpecContainer';
+
+const props = {
```
Comment (sachungo) [inline src/__test__/_components/AddAssetSpecContainer.test.js]: Why is props defined in the global scope?

File: src/components/common/ModalComponent.jsx [hunk 2]
```diff
@@ -77,7 +80,7 @@ ArtModal.defaultProps = {
   modalTitle: '',
   modalSize: 'small',
   open: false,
-  closeOnEscape: true,
-  closeOnDimmerClick: true,
+  closeOnEscape: false,
+  closeOnDimmerClick: false,
```
Comment (sachungo) [inline src/components/common/ModalComponent.jsx]: I hope you have tested the change that introduced the `true` value for these...I think it was the session expired modal

File: src/_components/AssetSpecs/AddAssetSpecContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    }
+  };
+
+  handleInputChange = (event, data = {}) => {
+    const name = event.target.name || data.name;
+    const value = event.target.value || data.value;
+
+    this.setState({
+      assetSpec: {
+        ...this.state.assetSpec,
```
Comment (sachungo) [inline src/_components/AssetSpecs/AddAssetSpecContainer.jsx]: If a new state depends on the previous state,  I recommend using an [updater function](https://reactjs.org/docs/react-component.html#setstate) as setting state is an asynchronous action. E.g
```jsx
this.setState(prevState => {
  return {
    ...prevState.assetSpec
   [name]: value
  }
})
```

File: src/__test__/_components/AddAssetSpecContainer.test.js (hunk 1 above)
Comment (Arthur236) [reply src/__test__/_components/AddAssetSpecContainer.test.js]: This was an oversight. Changing it now

File: src/components/common/ModalComponent.jsx (hunk 2 above)
Comment (Arthur236) [reply src/components/common/ModalComponent.jsx]: I had changed it to test something out then forgot to revert the values.

**Pull request merged**